### PR TITLE
feat (collections): add `runningReduce`

### PR DESCRIPTION
**Pull request by ayame113**
last part of #1173 

- [x] Did you support importing from mod.ts as well as your functions file?
- [x] Have you made sure to allocate as little memory and do as little steps in your algorithm as possible?
- [x] Did you add/adapt JSDoc comments, add/adapt an example, made sure that the example uses the ts markdown tag and assertEquals to show the result?
- [x] Did you add/adapt your functions documentation in README.md?
- [x] Did you make sure not to mutate the arguments passe to your function?
- [x] Did you add tests ensuring no mutation, empty input and corner cases?
- [x] Are your types flat, meaning there is no unnecessary alias in them that makes your users "go to type definition" twice to understand what to pass?

### CODE REVIEW

File: collections/running_reduce.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ * import { assertEquals } from "https://deno.land/std@$STD_VERSION/testing/asserts.ts";
+ *
+ * const numbers = [1, 2, 3, 4, 5];
+ * const sumSteps = runningReduce(numbers, (sum, current) => sum + current, 0);
+ *
+ * assertEquals(sumSteps, [1, 3, 6, 10, 15]);
+ * ```
+ */
+export function runningReduce<T, A>(
+  collection: readonly T[],
```
Comment (LionC) [inline collections/running_reduce.ts:21]: We have settled on `array` instead of `collection` for this argument in all other functions, I think we should be consistent here :-)

File: collections/running_reduce.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+ * import { runningReduce } from "https://deno.land/std@$STD_VERSION/collections/mod.ts";
+ * import { assertEquals } from "https://deno.land/std@$STD_VERSION/testing/asserts.ts";
+ *
+ * const numbers = [1, 2, 3, 4, 5];
+ * const sumSteps = runningReduce(numbers, (sum, current) => sum + current, 0);
+ *
+ * assertEquals(sumSteps, [1, 3, 6, 10, 15]);
+ * ```
+ */
+export function runningReduce<T, A>(
```
Comment (LionC) [inline collections/running_reduce.ts:20]: Very minor nitpick: If there is only one output type, it is called `O` in the other functions (I think consistently).

File: collections/running_reduce.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  collection: readonly T[],
+  reducer: (accumulator: A, current: T) => A,
+  initialValue: A,
+): A[] {
+  let _initialValue = initialValue;
```
Comment (LionC) [inline collections/running_reduce.ts:25]: Minor nitpick: Underscore prefixed values have some historical meaning attached to them which does not really apply here. I would just choose some "normal" name (like `currentResult` or something?)

File: collections/running_reduce.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  return collection.map((el) => _initialValue = reducer(_initialValue, el));
```
Comment (LionC) [inline collections/running_reduce.ts:26]: This is actually pretty smart!

I would personally put the function body on a newline, as a lot of things are happening here, the line is hard enough to read as it is and it hides a statements in there.

```suggestion
  return collection.map((el) =>
    _initialValue = reducer(_initialValue, el)
  );
```

**Review by LionC — CHANGES_REQUESTED** [review]
Thanks for implementing this! Loos good, I just added naming and API comments for consistency with the module. I really like the implementation.

**Comment by ayame113** [comment]
@LionC 
Thank you for your review, I have updated the PR.

File: collections/running_reduce.ts (hunk 4 above)
Comment (ayame113) [reply collections/running_reduce.ts]: When I apply this change, it doesn't seem to pass the `deno fmt` check.

**Review by LionC — APPROVED** [review]
LGTM! Pinging @kt3k

**Review by kt3k — APPROVED** [review]
@ayame113 LGTM. Nice work!

@LionC Thank you for reviewing in detail
